frontend/public/components/monitoring/receiver-forms/pagerduty-receiver-form.tsx

```diff
@@ -0,0 +1,76 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { SectionHeading } from '../../utils';
+import { RadioInput } from '../../radio';
+
+export const Form = ({ formValues, handleChange }) => (
+  <div data-test-id="pagerduty-receiver-form" className="co-m-pane__body--section-heading">
+    <SectionHeading text="PagerDuty Configuration" />
+    <div className="form-group">
+      <label className="control-label">Integration Type</label>
+      <div>
+        <RadioInput
+          title="Events API v2"
+          name="pagerDutyIntegrationType"
+          id="integration-type-events"
+          value="events"
+          onChange={handleChange}
+          checked={formValues.pagerDutyIntegrationType === 'events'}
+          inline
+        />
+        <RadioInput
+          title="Prometheus"
+          name="pagerDutyIntegrationType"
+          data-test-id="integration-type-prometheus"
+          value="prometheus"
+          onChange={handleChange}
+          checked={formValues.pagerDutyIntegrationType === 'prometheus'}
+          inline
+        />
+      </div>
+    </div>
+    <div className="form-group">
+      <label data-test-id="pagerduty-key-label" className="control-label co-required">
+        {formValues.pagerDutyIntegrationType === 'events' ? 'Routing' : 'Service'} Key
+      </label>
+      <input
+        className="pf-c-form-control"
+        type="text"
+        aria-describedby="integration-key-help"
+        name="pagerDutyIntegrationKey"
+        data-test-id="integration-key"
+        value={formValues.pagerDutyIntegrationKey}
+        onChange={handleChange}
+      />
+      <div className="help-block" id="integration-key-help">
+        PagerDuty integration key
+      </div>
+    </div>
+  </div>
+);
+
+export const getInitialValues = (receiverConfig) => {
+  return _.isEmpty(receiverConfig)
+    ? {
+        pagerDutyIntegrationType: 'events', // 'Events API v2'
+        pagerDutyIntegrationKey: '',
+      }
+    : {
+        pagerDutyIntegrationType: _.get(receiverConfig, 'routing_key') ? 'events' : 'prometheus',
```

Comment:
```suggestion
        pagerDutyIntegrationType: receiverConfig?.routing_key ? 'events' : 'prometheus',
```

Reply:
Has to be `receiverConfig?.['routing_key']` else I get camel-case eslint error

Reply:
that defeats the purpose of using optional chaining :) better to add an eslint ignore. if it's present in many places in a file, it would be reasonable to add it at the file level

Reply:
ok, fixed